# ANALYZE OF APP STORES INTEREST

In this project we have two data sets, which are obtained from App Store and Google Play Store. Our aim is understanding the interest of users on an application to determine our road map for next application projects. We will analyze the free apps which are in English. For free apps, we need to understand number of users.

You can download data sets from their source, if you want to download. 
<br>
[App Store data set](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)
<br>
[Google Play Store data set](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv) 

## EXPLORING DATA SETS

In [1]:
from csv import reader
opened_file1= open("googleplaystore.csv", encoding="utf8")
opened_file2= open("AppleStore.csv", encoding="utf8")
read_file1= reader(opened_file1)
read_file2= reader(opened_file2)

In [2]:
#Apple App Store data set
ios_apps= list(read_file2)
ios_header= ios_apps[0]
ios= ios_apps[1:]

#Google Play Store dataset
android_apps= list(read_file1)
android_header= android_apps[0]
android= android_apps[1:]



In [3]:
#Function to explore data sets
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print ('Number of rows: ', len(dataset))
        print ('Number of columns: ', len(dataset[0]))
        
        

In [4]:
explore_data(ios,2,6,True)

['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


Number of rows:  7197
Number of columns:  16


In [5]:
explore_data(android,0,10,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+

 We have 7197 IOS apps in this data set, and the columns. If you want to learn more details about data set, it can be found in the data set [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

Also, we have 10841 apps in Google Play data set. To obtain details about each column, you can use [this link](https://www.kaggle.com/lava18/google-play-store-apps).

## CLEANING DATA SETS

### Cleaning Wrong Data

Before deep into analyzing data sets, we have to be sure that our data is clean. [In duscussion page](https://www.kaggle.com/lava18/google-play-store-apps/discussion) about Google Play Store data set, we see an error on [one of the discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015). Let's look at this wrong data, and clean it.

In [6]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [7]:
len(android)
del (android[10472])
len(android)

10840

### Cleaning Duplicate Data

Now we have 10840 apps to analyze in Google App Store data set. But, we can see that there are some duplicate apps, which can effect to our analyze wrongly. You can see this fact from apps which name is "Instagram" below, as an example. 

In [8]:
for app in android:
    name=app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [9]:
#Function to detect duplicate apps in Google Play Store data set
duplicate_apps=[]
unique_apps= []

for app in android:
    name= app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])


Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


In [10]:
#Function to detect duplicate apps in Apple App Store data set
duplicate_ios=[]
unique_ios=[]

for app in ios:
    name= app[1]
    if name in unique_ios:
        duplicate_ios.append(name)
    else:
        unique_ios.append(name)
    
print("The number of duplicate ios apps: ", len(duplicate_ios))
print(duplicate_ios)

The number of duplicate ios apps:  2
['Mannequin Challenge', 'VR Roller Coaster']


We have 2 duplicate apps in App Store data set and 1181 duplicate apps in Google Play Store. Let's clean exstra apps.

In [11]:
#Duplicate apps in App Store
for app in ios: 
    name=app[1]
    if name == 'Mannequin Challenge' or name=='VR Roller Coaster' :
        print(app)

['1173990889', 'Mannequin Challenge', '109705216', 'USD', '0.0', '668', '87', '3.0', '3.0', '1.4', '9+', 'Games', '37', '4', '1', '1']
['952877179', 'VR Roller Coaster', '169523200', 'USD', '0.0', '107', '102', '3.5', '3.5', '2.0.0', '4+', 'Games', '37', '5', '1', '1']
['1178454060', 'Mannequin Challenge', '59572224', 'USD', '0.0', '105', '58', '4.0', '4.5', '1.0.1', '4+', 'Games', '38', '5', '1', '1']
['1089824278', 'VR Roller Coaster', '240964608', 'USD', '0.0', '67', '44', '3.5', '4.0', '0.81', '4+', 'Games', '38', '0', '1', '1']


In [12]:
#An example of duplicate apps in Google Store
for app in android:
    name= app[0]
    if name == 'Google My Business':
        print(app)

['Google My Business', 'BUSINESS', '4.4', '70991', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 24, 2018', '2.19.0.204537701', '4.4 and up']
['Google My Business', 'BUSINESS', '4.4', '70991', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 24, 2018', '2.19.0.204537701', '4.4 and up']
['Google My Business', 'BUSINESS', '4.4', '70991', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 24, 2018', '2.19.0.204537701', '4.4 and up']


Now, we have to decide how we can select the true app from this duplicate apps. Actually, we can select the app which has more reviews. Because, the high review can show the last correction on app data.

In [13]:
#Show the amount of the maximum reviews for duplicate apps in Google Play data set 
reviews_max={}
for app in android:
    name= app[0]
    n_reviews= float(app[3])
    if name in reviews_max and reviews_max[name]< n_reviews:
        reviews_max[name] = n_reviews
    if name not in reviews_max:
        reviews_max[name] = n_reviews
        
print('The expected length:', len(android)-1181)
print('The actual length:', len(reviews_max))

The expected length: 9659
The actual length: 9659


In [14]:
#Show the amount of the maximum rewievs for duplicate apps in App Store data set
user_max={}
for app in ios:
    name= app[1]
    rating_count= float(app[5])
    if name in user_max and user_max[name]<rating_count:
        user_max[name]= rating_count
    else:
        user_max[name]= rating_count
print('The expected length:',len(ios)-2 )        
print('The actual length:',len(user_max))

The expected length: 7195
The actual length: 7195


When we evaluate the results, we see that it is compatible with the data we expect. We can use this method to get rid of repeated apps.

In [15]:
#Google Play data set without duplicate apps
android_clean=[]
already_added=[]

for app in android:
    name= app[0]
    n_reviews= float(app[3])
    if (reviews_max[name]== n_reviews) and name not in already_added:
        android_clean.append(app)
        already_added.append(name)
        

print(len(android_clean))
                

9659


In [16]:
#App Store data set without duplicate apps
ios_clean=[]
already_addedd=[]

for app in ios:
    name= app[1]
    rating_count= float(app[5])
    if (user_max[name]== rating_count) and name not in already_addedd:
        ios_clean.append(app)
        already_added.append(name)
        

print(len(ios_clean))
        

7195


In this step we found the data which is cleaned from duplicates.

### Cleaning Non-English Apps Data

Our plan is to make decision on apps which is in English. However we have non-English apps, and we should get rid of this apps to analyze correctly. You can find some examples: 

In [17]:
print('App Store Non-English Examples')
print(ios[813][1])
print(ios[6731][1])
print("\n")
print('Google Play Non-English Examples')
print(android_clean[4412][0])
print(android_clean[7940][0])

App Store Non-English Examples
爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


Google Play Non-English Examples
中国語 AQリスニング
لعبة تقدر تربح DZ


For English alphabet, we know that, they have ASCII values which are from 0 to 127. So, we can use this information to detect apps in English.

In [18]:
#Find Apps in English
def english_detector(value):
    
    for character in value:
        
        if ord(character)>127:
            return False
        
    return True
            

In [19]:
english_detector('Instagram')

True

In [20]:
english_detector('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [21]:
english_detector('Docs To Go™ Free Office Suite')

False

In [22]:
english_detector('Instachat 😜')

False

In [23]:
ord('😜')

128540

Right now, we have a problem. We never accept the special characters in first function. However, there are some apps in English, although they have special character, as above. So, we should put in mind these characters to hinder data loss.And we should develop our function according to these special characters.

In [24]:
#New version detector function
def is_english(value):
    non_english=0
    for character in value:
      
        if ord(character)>127:
            non_english+=1
            
    if non_english>3:
        return False
    else:
        return True

In [25]:
is_english('Docs To Go™ Free Office Suite')

True

In [26]:
is_english('Instachat 😜')

True

In [27]:
is_english('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

No longer, we can use the function on our data sets to get apps in English. It is not a perfect selection, but useful fot proper analyze. 

In [28]:
#Find Apps in English in Google Play Store
android_english=[]

for app in android_clean:
    name= app[0]
    if is_english(name):
        android_english.append(app)
        
print(explore_data(android_english,0,6,True))


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50,000+

In [29]:
#Find Apps in English in App Store
ios_english=[]

for app in ios_clean:
    name= app[1]
    if is_english(name):
        ios_english.append(app)
        
print(explore_data(ios_english,0,6,True))

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


Number of rows:  6181
Number of columns:  16
None


## SEPERATE FREE APPS

Our plan is in this project analyze the free apps to use in our next projects. So, we need to seperate the free apps from out core data sets. We will obtain our last data set to analyze. 

In [30]:
#Find the free apps from Google Play Store data set
android_final= []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print('The lenght of final Google Play Store data set:' ,len(android_final))
print('\n')
print('The lenght of final App Store data set:',len(ios_final))

The lenght of final Google Play Store data set: 8864


The lenght of final App Store data set: 3220


## ANALYZE DATA SETS TO DECISION

Now we have our final data sets. With these data sets, we can understand the interest of users. Our final goal is to find proper app profile which is succesful in two platforms.

After this part, we can start with calculating the frequencies different types of the apps.

In [31]:
def freq_table(dataset, index):
    table={}
    total=0
    
    for row in dataset:
        name= row[index]
        total+=1
        if name in table:
            table[name]+=1
        else:
            table[name]=1
    

    per_table={}
    for key in table:
        percentage= (table[key]/total)*100
        per_table[key]=percentage
        
    return per_table
    
            

In [32]:
def display_table(dataset, index):
    table= freq_table(dataset,index)
    table_display=[]
    
    for key in table:
        key_val_as_tuple= (table[key], key)
        table_display.append(key_val_as_tuple)
    
    table_sorted= sorted(table_display, reverse= True)
    for entry in table_sorted:
        print(entry[1],':', entry[0])

In [33]:
print('App Store columns')
print(ios_header)
print('\n')
print('Google Store columns')
print(android_header)


App Store columns
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Google Store columns
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


With the help of the column names above, we can start examining the distributions.
<br>
### Genres
<br>
Let's start examining with exploring frequencies of genres.

In [34]:
display_table(ios_final,-5 ) #App Store Frequencies of Genres

Games : 58.13664596273293
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.291925465838509
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


If we want to talk about App Store apps, we see that the genre of 'Games' has a big interest, that is 58%. The second genre is 'Entertainment', that is 7.8%. And the third one is 'Photo& Video',that is 4.9%. Also, there are more application that they has close frequencies to each other.  

In [35]:
display_table(android_final,-4) #Google Play Store Frequencies of Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

In [36]:
display_table(android_final,1)# Google Play Store Frequencies of Categories

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

There is interesting results in Google Play, we can not see the apps at the top of the apps which is made for 'game' or 'entertainment'. But these catogories close to top. Trying to analyze the app profiles from Google Play store with using the genres to analyze can be complicated, because the genres are much granular. To obtain a general results we will use categories.   

## FINDING MOST POPULAR APPS ACCORDING TO GENRES ON APP STORE

Now, we are close to our aim. We can find the most popular apps to understand interest of users. We will use frequency table again, it will show us our target genre and we will see most popular apps in App Store data set. 

In [37]:
ios_genre= freq_table(ios_final,11)

#To calculate average number of user ratings per app genre on App Store
for genre in ios_genre:
    total=0
    len_genre= 0
    for app in ios_final:
        genre_ios= app[11]
        if genre_ios== genre:
            n_ratings= float(app[5])
            total+= n_ratings
            len_genre+=1
    avg_n_rating=(total/len_genre)
    print(genre, ':', avg_n_rating)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22812.602564102563
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


We see that 'Navigation' has the highest number of user ratings.

In [38]:
# To show the number of rating on 'Navigation' genre
for app in ios_final:
    if app[-5]=='Navigation':
        print(app[1],':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


'Waze - GPS Navigation, Maps & Real-time Traffic' has the highest reviews.
<br>
The second genre is 'Reference' which has highest number of user ratings.

In [39]:
#To show the number of rating on 'Reference' genre
for app in ios_final:
    if app[-5]=='Reference':
        print(app[1], ":", app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


'Bible' has the highest number of user reviews, and the second one is 'Dictionary.com Dictionary & Thesaurus'. 

##FINDING MOST POPULAR APPS ACCORDING TO CATEGORIES ON GOOGLE PLAY STORE

We have number of installs to use. However, this data is not precise to calculate.

In [40]:
display_table(android_final,5) #Show the shape of data

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We can start with cleaning this data to make it precise and calculate the average number of the categories.

In [41]:
freq_android= freq_table(android_final,1)

for category in freq_android:
    total=0
    len_total=0
    
    for app in android_final:
        category_app= app[1]
        if category_app== category:
            install= app[5]
            install= install.replace(',','')
            install=install.replace('+','')
            total+= float(install)
            len_total+=1
    avg_n_install= total/ len_total
    print(category, ':', avg_n_install)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

'Communication' category has the highest number of installs.

In [42]:
for app in android_final:
    if app[1]=="COMMUNICATION":
        print (app[0], ":", app[5])
    

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

There are a lot of application under the communication category. We can select the most popular ones. 

In [43]:
#To find the apps which have (100,000,000; 500,000,000; 1,000,000,000) installs
for app in android_final:
    if app[1]=='COMMUNICATION' and (app[5]=='1,000,000,000+'
                                    or app[5]=='500,000,000+'
                                    or app[5]=='100,000,000+'):
        print (app[0], ":", app[5])
    

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

If we ignore the communication apps which have installs over 100 million, the average will reduce roughly ten times:

In [44]:
under_100_m=[]

#To ignore the communication apps which have installs over 100 million
for app in android_final:
    installs= app[5]
    installs= installs.replace(',','')
    installs= installs.replace('+','')
    if app[1]=='COMMUNICATION' and (float(installs)< 100000000):
        under_100_m.append(float(installs))
        
    
sum(under_100_m)/ len(under_100_m)   

3603485.3884615386

This average is more proper for us, because it is to difficult to compete with popular communication apps for a new application project. The second category is 'BOOKS_AND_REFERENCE' which has high number of installs.

In [45]:
for app in android_final:
    if app[1]=='BOOKS_AND_REFERENCE':
        print (app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

As we can see, there are some apps which has 100,000,000 and more installs.

In [46]:
#To show the books and references apps which have highest installs
for app in android_final:
    if app[1]=='BOOKS_AND_REFERENCE' and (app[5] =='1,000,000,000+'
                                         or app[5]=='500,000,000+'
                                         or app[5]=='100,000,000+'):
        print(app[0], ":", app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


There is not too much insuperable apps in this field than communication apps. When we check the number of installs, we can see that the users have interest to install book & reference apps. 